service/controller/resource/appversionlabel/create.go

```diff
@@ -8,96 +8,133 @@ import (
 	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
 	"github.com/giantswarm/apiextensions/v6/pkg/label"
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	apiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
 
-	"github.com/giantswarm/cluster-operator/v5/pkg/project"
 	"github.com/giantswarm/cluster-operator/v5/service/controller/key"
 	"github.com/giantswarm/cluster-operator/v5/service/internal/releaseversion"
 
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
-	cr, err := key.ToCluster(obj)
+	cluster, err := key.ToCluster(obj)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	var apps []*v1alpha1.App
-	{
-		r.logger.Debugf(ctx, "finding optional apps for tenant cluster %#q", key.ClusterID(&cr))
+	logger := r.logger.With("cluster", key.ClusterID(&cluster))
 
-		o := metav1.ListOptions{
-			LabelSelector: fmt.Sprintf("%s!=%s", label.ManagedBy, project.Name()),
+	apps, err := r.getApps(ctx, logger, cluster)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.updateApps(ctx, logger, cluster, apps)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func (r *Resource) getApps(ctx context.Context, logger micrologger.Logger, cluster apiv1beta1.Cluster) ([]*v1alpha1.App, error) {
+	logger.Debugf(ctx, "finding apps")
+
+	list := &v1alpha1.AppList{}
+	err := r.ctrlClient.List(ctx, list, &client.ListOptions{Namespace: key.ClusterID(&cluster), Raw: &metav1.ListOptions{}})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	apps := []*v1alpha1.App{}
+	for _, item := range list.Items {
+		apps = append(apps, item.DeepCopy())
+	}
```

Comment:
Wouldn't this be the same as returning `list.Items` without looping?

Reply:
It actually converts it from `[]v1alpha1.App` to `[]*v1alpha1.App`. I suppose for convenience later as pointers are neede